### PR TITLE
Modifies Vagrantfile and Vagrant provisioning scripts. Some scripts n…

### PR DESCRIPTION
**Pull request by Xavi5**
Creating provision environment variables form scritps not working.

Some commands not available when launched from provisioning scripts
==> main_app: /tmp/vagrant-shell: line 13: bundle: command not found
==> main_app: /tmp/vagrant-shell: line 6: rvm: command not found

### CODE REVIEW

**Comment by ryanfox1985** [comment]
Hi Xavi,

Please discard the Gemfile.lock changes.
Please not modify these files:
- https://rvm.io/integration/vagrant

Your Vagrantfile provision looks disordered, can you refactor the order to:
- Install RVM (vagrant/install-rvm.sh).
- Install Ruby 2.3.0 (vagrant/install-ruby.sh).
- Install project deps, software needed, ... (vagrant/setup.sh).
- Start project (vagrant/start.sh) (Run always).

RVM uses two files ruby-gemset and ruby-version can you add?

Regards.

**Comment by ryanfox1985** [comment]
Xavi,

use the script vagrant/install-ruby.sh to install ruby version. And if you want you can install commons gems like 'rails, bundler'

Regards.

**Comment by ryanfox1985** [comment]
Hi Xavi,

it's better now, but the good order are:
- install rvm [not privileged].
- install ruby (only the version 2.3.0 and default gems rails) [not privileged].
- setup
- start (provision: always) [not privileged]

#### install-rvm

If the command gpg from install-rvm.sh gives you a timeout you can use another key server:

```
gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
gpg --keyserver hkp://pgp.mit.edu --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
```

#### install-ruby

For install ruby [install-ruby.sh] use the next file:

```
#!/bin/bash

source $HOME/.rvm/scripts/rvm

rvm use --default --install $1
echo "gem: --no-rdoc --no-ri" > ~/.gemrc

shift

if (( $# ))
    then
    rvm use $1@global
    gem install $@
fi

rvm cleanup all
```

#### setup

Install project dependencies:
- apk-get
- nodejs
- mailcatcher (you can use ruby 2.3.0).
- bundle install
- create database; schema load; ecommerce initializations.

#### start
- bundle install
- migrations

After vagrant up, the box it's ready to enter by ssh, enter to /vagrant and start as: 

```
rails s Puma -b 0.0.0.0
```

Regards.

**Comment by Xavi5** [comment]
Hi ryanfox,

I tried to fit your directions. Now ylosix demo is up just by running 'rails s -b 0.0.0.0' on /vagrant just after doing 'vagrant ssh' without any errors.

I keep the install-ruby.sh and install-rvm.sh scripts unmodified as you want, but it is needed to run "apt-get install -qy curl" before install-rvm.sh, so I provision setup.sh before in Vagrantfile and run all apt.-get there.

After installing rvm and ruby, I provisión another setup script whith commands needed to run after ruby install: install gems, create db, set environment variables...

I had problems provisioning environment variables using vagrant provisioner scripts. I implemented a solution in set_env_var.sh as explained here:
http://stackoverflow.com/questions/24707986/create-linux-environment-variable-using-vagrant-provisioner

Regards.

File: vagrant/setup.sh [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-# Removed if already prepared
-rm -rf ./tmp/container_prepared
-
-# Run and link the containers
+# Run postgres container
 docker run -d -p 5432:5432 -v /vagrant:/vagrant -e "POSTGRES_PASSWORD=postgres" --name postgres postgres:9.4.1
-docker run -d -p 3000:3000 -v /vagrant:/var/www --privileged --link postgres:db \
-        -e "RAILS_ENV=development" \
-        -e "DATABASE_URL=postgres://postgres:postgres@db:5432/ecommerce" \
-        --name rails ylosix/ylosix ./vagrant/start_container.sh
```
Comment (ryanfox1985) [inline vagrant/setup.sh]: Put apt-get install all in one line.

Comment (ryanfox1985) [inline vagrant/setup.sh]: I think that it's a good idea to install postgresql-client in the development box.

File: Vagrantfile [hunk 2]
```diff
@@ -24,11 +24,15 @@ Vagrant.configure(2) do |config|
     # Ubuntu
     app.vm.box = 'box-cutter/ubuntu1404-docker'
 
-    # Setup the containers when the VM is first created
+    # install RVM
+    config.vm.provision :shell, path: "vagrant/install-rvm.sh", args: "stable", privileged: false
+    # install Ruby
+    config.vm.provision :shell, path: "vagrant/install-ruby.sh", args: "2.3.0 rails bundler", privileged: false
```
Comment (adrianmo) [inline Vagrantfile]: This should be `app`

File: Vagrantfile [hunk 3]
```diff
@@ -24,11 +24,15 @@ Vagrant.configure(2) do |config|
     # Ubuntu
     app.vm.box = 'box-cutter/ubuntu1404-docker'
 
-    # Setup the containers when the VM is first created
+    # install RVM
+    config.vm.provision :shell, path: "vagrant/install-rvm.sh", args: "stable", privileged: false
```
Comment (adrianmo) [inline Vagrantfile]: This should be `app`

File: Vagrantfile [hunk 4]
```diff
@@ -24,11 +24,15 @@ Vagrant.configure(2) do |config|
...
+    # install Ruby
+    config.vm.provision :shell, path: "vagrant/install-ruby.sh", args: "2.3.0 rails bundler", privileged: false
+    # Setup project dependencies and postgres container
     app.vm.provision 'shell', path: 'vagrant/setup.sh'
-
-    # Make sure the correct containers are running
-    # every time we start the VM.
-    app.vm.provision 'shell', run: 'always', path: 'vagrant/start.sh'
+    #set environment variables
+    config.vm.provision 'shell', run: 'always', path: "vagrant/set_env_var.sh", privileged: false
```
Comment (adrianmo) [inline Vagrantfile]: This should be `app`

File: vagrant/install-ruby.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
```
Comment (adrianmo) [inline vagrant/install-ruby.sh]: Change this script by the one @ryanfox1985 suggested on the PR

File: Vagrantfile (hunk 4 above)
Comment (adrianmo) [inline Vagrantfile]: Well, this script should be included in the `start.sh` script.

File: vagrant/set_env_var.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Sets needed environment variables
+source ~/.bashrc
+
+if [ -z "$RAILS_ENV" ]; then # only checks if VAR is set, regardless of its value
+    echo "export RAILS_ENV=development" >> ~/.bashrc
```
Comment (adrianmo) [inline vagrant/set_env_var.sh]: This appends the ENV var every time there is a `vagrant up`. Wrong behaviour.

File: vagrant/set_env_var.sh [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+fi
+
+if [ -z "$DATABASE_URL" ]; then # only checks if VAR is set, regardless of its value
+    echo "export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce" >> ~/.bashrc
```
Comment (adrianmo) [inline vagrant/set_env_var.sh]: Same here

File: vagrant/setup_gems_database.sh [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+echo "gem: --no-rdoc --no-ri" >> ~/.gemrc
+gem install bundler
+
+#create and use a developmet gemset
+rvm use ruby-2.3.0@development --create
+rvm --default use ruby-2.3.0@development
+
+# Set the working environment
+export RAILS_ENV=development
+export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce
```
Comment (adrianmo) [inline vagrant/setup_gems_database.sh]: These env vars are exported in multiple places. Revisit that.

File: vagrant/start.sh [hunk 9]
```diff
@@ -1,8 +1,16 @@
 #!/bin/bash
 
-# Commands required to ensure correct docker containers are started when the vm is rebooted.
+# TODO Set the working environment
+#export RAILS_ENV=development
```
Comment (adrianmo) [inline vagrant/start.sh]: Clean up unused code

File: vagrant/setup.sh [hunk 10]
```diff
@@ -1,22 +1,39 @@
...
+echo "gem: --no-rdoc --no-ri" >> ~/.gemrc
+gem install bundler
+
+# Set the working environment
+source ~/.profile && [ -z "$DATABASE_URL" ] && \
+    echo "export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce" >> ~/.profile
+
+# generate database
+cd /vagrant
+bundle install
```
Comment (ryanfox1985) [inline vagrant/setup.sh]: Rewrite these line as:
cd /vagrant; bundle install
cd /vagrant; bundle exec rake db:create db:schema:load db:gen_demo

Same in the start.sh

File: vagrant/start.sh [hunk 11]
```diff
@@ -1,8 +1,11 @@
...
+
+# install new gems and make new migrations
+cd /vagrant
+bundle install
+bundle exec rake db:migrate
+
+# Commands required to ensure correct docker containers are started when the vm is rebooted.
+docker start postgres
+
+#mailcatcher --ip 0.0.0.0
```
Comment (ryanfox1985) [inline vagrant/start.sh]: uncomment this line.

File: vagrant/setup.sh [hunk 12]
```diff
@@ -1,22 +1,39 @@
...
+sudo apt-get install -qy imagemagick
+#sudo gem install mailcatcher
+sudo apt-get install -qy libpq-dev   # dependency for pg gem
+sudo apt-get install -qy postgresql-client
+
+# To start using RVM you need to run `source /home/vagrant/.rvm/scripts/rvm`
+# in all your open shell windows, in rare cases you need to reopen all shell windows.
+source /home/vagrant/.rvm/scripts/rvm
+rm -rf /usr/local/bundle/config
+
```
Comment (ryanfox1985) [inline vagrant/setup.sh]: remove these lines:
+gem update --system
 +rvm gemset use global
 +gem update
 +# disable gem documentation install to speed up gem installation
 +echo "gem: --no-rdoc --no-ri" >> ~/.gemrc
 +gem install bundler

File: vagrant/install-ruby.sh [hunk 13]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+source $HOME/.rvm/scripts/rvm
+
+rvm use --default --install $1
+echo "gem: --no-rdoc --no-ri" > ~/.gemrc
```
Comment (ryanfox1985) [inline vagrant/install-ruby.sh:7]: Missing this: Add these comment:

```
# disable gem documentation install to speed up gem installation
echo "gem: --no-rdoc --no-ri" > ~/.gemrc
```

File: vagrant/setup.sh [hunk 14]
```diff
@@ -1,22 +1,39 @@
...
+# Stop and remove any existing containers
+docker stop $(docker ps -a -q)
+docker rm $(docker ps -a -q)
+
+# Run postgres container
+docker run -d -p 5432:5432 -v /vagrant:/vagrant -e "POSTGRES_PASSWORD=postgres" --name postgres postgres:9.4.1
+
+# install required packages
+sudo apt-get update
+sudo apt-get install -qy git
```
Comment (ryanfox1985) [inline vagrant/setup.sh]: All apt-get lines in one.

File: vagrant/setup.sh [hunk 15]
```diff
@@ -1,22 +1,39 @@
...
+sudo apt-get -qy install nodejs
+sudo apt-get install -qy imagemagick
+#sudo gem install mailcatcher
```
Comment (ryanfox1985) [inline vagrant/setup.sh]: Add mailcatcher at Vagrantfile:

```
app.vm.provision :shell, path: "vagrant/install-ruby.sh", args: "2.3.0 rails bundler", privileged: false
```

File: vagrant/setup-app.sh [hunk 16]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+# generate database
+cd /vagrant; bundle install; bundle exec rake db:create db:schema:load db:gen_demo
```
Comment (ryanfox1985) [inline vagrant/setup-app.sh]: Put this in two lines:

``````
cd /vagrant; bundle install 
cd /vagrant; bundle exec rake db:create db:schema:load db:gen_demo
```_
``````

File: vagrant/setup-env.sh [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+docker stop $(docker ps -a -q)
+docker rm $(docker ps -a -q)
+
+# Run postgres container
+docker run -d -p 5432:5432 -v /vagrant:/vagrant -e "POSTGRES_PASSWORD=postgres" --name postgres postgres:9.4.1
+
+# install required packages
+sudo apt-get update
+sudo apt-get install -qy git nodejs imagemagick libpq-dev postgresql-client
+    #libpq-dev   # dependency for pg gem
```
Comment (ryanfox1985) [inline vagrant/setup-env.sh]: Remove this comment:

```
#libpq-dev   # dependency for pg gem
```

File: vagrant/start.sh [hunk 18]
```diff
@@ -1,8 +1,9 @@
...
-
-docker start postgres
-docker start rails
-
-mailcatcher --ip 0.0.0.0
\ No newline at end of file
+#!/bin/bash
+
+# install new gems and make new migrations
+cd /vagrant; bundle install; bundle exec rake db:migrate
```
Comment (ryanfox1985) [inline vagrant/start.sh]: Put this in two lines:

```
cd /vagrant; bundle install
cd /vagrant; bundle exec rake db:migrate
```

File: Vagrantfile [hunk 19]
```diff
@@ -1,34 +1,45 @@
...
+    app.vm.box = 'box-cutter/ubuntu1404-docker'
+
+    # Set environment variables
+    $script = <<-SHELL
+          source ~/.profile && [ -z "$DATABASE_URL" ] && echo "export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce" >> ~/.profile
+          source ~/.profile && [ -z "$RAILS_ENV" ] && echo "export RAILS_ENV=development" >> ~/.profile
+    SHELL
+    app.vm.provision "shell", privileged: false, inline: $script
+    # set rvm project config files
+    app.vm.provision "shell", inline: "echo 2.3.0 > .ruby-version", privileged: false
```
Comment (ryanfox1985) [inline Vagrantfile]: Add this in the shell script that appears before.

File: Vagrantfile [hunk 20]
```diff
@@ -1,34 +1,45 @@
...
+    app.vm.provision "shell", inline: "echo ylosix > .ruby-gemset", privileged: false
```
Comment (ryanfox1985) [inline Vagrantfile]: Add this in the shell script that appears before.

File: Vagrantfile [hunk 21]
```diff
@@ -1,34 +1,44 @@
...
+    # Ubuntu
+    app.vm.box = 'box-cutter/ubuntu1404-docker'
+
+    # Set environment variables
+    app.vm.provision "shell", privileged: false, inline: <<-SHELL
+      source ~/.profile && [ -z "$DATABASE_URL" ] && echo "export DATABASE_URL=postgres://postgres:postgres@127.0.0.1:5432/ecommerce" >> ~/.profile
+      source ~/.profile && [ -z "$RAILS_ENV" ] && echo "export RAILS_ENV=development" >> ~/.profile
+    SHELL
+    # set rvm project config files
+    app.vm.provision "shell", inline: "echo 2.3.0 > .ruby-version", privileged: false
```
Comment (ryanfox1985) [inline Vagrantfile]: Add this line in the shell block before.

File: Vagrantfile [hunk 22]
```diff
@@ -1,34 +1,44 @@
...
+    app.vm.provision "shell", inline: "echo ylosix > .ruby-gemset", privileged: false
```
Comment (ryanfox1985) [inline Vagrantfile]: Add this line in the shell block before.

File: README.md [hunk 23]
```diff
@@ -13,7 +13,7 @@ This repository contains the source for the latest version of Ylosix as well as
 
 ### Demo
 - [Frontend](http://ylos.ylosix.com)
-- [Backoffice](http://ylos.ylosix.com/admin) (email: `admin@example.com`, password: `password`)
+- [Backoffice](http://ylos.ylosix.com/admin) (email: `admin@ylos.com`, password: `password`)
```
Comment (ryanfox1985) [inline README.md]: The user is wrong:
https://github.com/ylosix/ylosix/blob/develop/db/seeds.rb#L32

**Pull request closed without merging**